src/public/app/room/room.service.ts

```diff
@@ -0,0 +1,50 @@
+import { Injectable } from 'angular2/core';
+import { Http } from 'angular2/http';
+import { PrivateRoom } from './PrivateRoom';
+import {
+    Room,
+    Song,
+    QueuedUpSong
+} from '../core/types';
+
+
+@Injectable()
+class RoomService {
+
+    private currentRoom: Room = null;
+    private allRooms: Room[] = [];
+
+    constructor(private http: Http){
+        //want to ping server for rooms, just testing with private room now
+        this.allRooms.push(new PrivateRoom());
```

Comment:
nit: I think it's clearer to have

this.allRooms = [new PrivateRoom()]; 

instead of declaring above and adding to it in the constructor.